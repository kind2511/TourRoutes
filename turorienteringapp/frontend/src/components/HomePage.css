/* Base styling for the HomePage component */

/* flex- home page will be two parts */
.home-page {
    display: flex;
    height: 100vh; /*full height*/
}

/* Style for the left side of  home page */
.home-page .left {
    width: 50%;
    background-color: lightgreen;
    padding: 100px 20px;
    text-align: left;
    position: relative;
}

/* Style for the right side of the home page */
.home-page .right {
    width: 50%;
    background-color: white;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center; /* Vertically centers the content in this div */
}

/* Paragraph styling for the sliding sentences */
.home-page p {
    font-size: 1.2rem;
    color: #4CAF50;
}

/* Styling for the logo (TurRuter) above the buttons */
.nav-logo {
    font-size: 2.9rem; /* Increase the logo's font size */
    margin-bottom: 10px; /* Add some space below the logo */
    color: #4CAF50; /* Green color to match the theme */
    text-decoration: none; /* No underlines for the logo text */
}

/* Styling for the buttons on the right side */
.home-page .right button {
    margin: 10px 20px; /* 20 for space between button og 10 for space with log*/
    padding: 15px 30px; /* 15 for height, 30 for wide   */
    font-size: 1.5rem;
    border: none;
    cursor: pointer;
    background-color: #45a049;
    color: white;
    width: 150px;
    text-align: center;
    transition: background-color 0.3s;
}

/* Hover effect for the buttons  */
.home-page .right button:hover {
    background-color: #61dafb;
    color: #333;
}

/* Styl3 for the sentence section on the left side */
.sentence-section {
    padding: 50px 0;
    position: absolute;
    top: 50%;
    left: 5%;
    transform: translateY(-50%);
    width: 90%;
}

.sentence-section p {
    font-size: 40px;
    color: #059b16;
}


/*
 *
 */
.home-page .right .footer-about {
    position: absolute;
    bottom: 10px;
    left: 1150px;
    font-size: 16px;
    text-decoration: none;
    text-decoration: none;
    color: #059b16;
    font-weight: bold;
}


/*  Link component styling*/
.home-page .right .footer-about a, 
.home-page .right .footer-about a:visited {
    text-decoration: none;
    color: #059b16; /* Dark green color */
}

/* Hover styling */
.home-page .right .footer-about a:hover {
    color: #1cf7f3;
}