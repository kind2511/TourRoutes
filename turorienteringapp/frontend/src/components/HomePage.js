import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import './HomePage.css';

/* An array of sentences that show on the left side of the homepage */
const sentences = [
    "Explore fascinating routes with TurRuter!",
    "Join us for an unparalleled adventure!",
    "Navigate, Discover, Experience!",
    "Your journey starts here!"
];

function HomePage() {
    /* Using React's useState to manage the current displayed text and the current sentence's index */
    const [currentText, setCurrentText] = useState('');
    const [currentSentenceIndex, setCurrentSentenceIndex] = useState(0);

    /* useEffect hook to manage the sliding effect of the sentences */
    useEffect(() => {
        let timer;
        if (currentText === sentences[currentSentenceIndex]) {
            timer = setTimeout(() => {
                setCurrentSentenceIndex((prevIndex) => (prevIndex + 1) % sentences.length);
                setCurrentText('');
            }, 3000);
        } else {
            timer = setTimeout(() => {
                setCurrentText(prev => sentences[currentSentenceIndex].substr(0, prev.length + 1));
            }, 50);
        }

        /* Clearing the timer to avoid any potential memory leaks */
        return () => clearTimeout(timer);
    }, [currentText, currentSentenceIndex]);

    /* Rendering the HomePage component */
    return (
        <div className="home-page">
            <div className="left">
                <div className="sentence-section">
                    <p>{currentText}</p>
                </div>
            </div>
            <div className="right">
                {/* TurRuter logo displayed above the buttons */}
                <Link to="/" className="nav-logo">TurRuter</Link>

                {/* Buttons section */}
                <div style={{ display: 'flex', justifyContent: 'center', marginTop: '20px' }}>
                    <button onClick={() => window.location.href="/login"}>Login</button>
                    <button onClick={() => window.location.href="/register"}>Register</button>
                </div>
            </div>
        </div>
    );
}

export default HomePage;
