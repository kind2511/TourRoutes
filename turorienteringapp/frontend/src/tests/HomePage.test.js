import React from 'react';
import { render, screen } from '@testing-library/react';
import HomePage from '../components/HomePage';
import { BrowserRouter } from 'react-router-dom';


test('renders HomePage component without crashing', () => {
  render(
    <BrowserRouter>
      <HomePage />
    </BrowserRouter>
  );
  
});

//--------------------------------------------->

// Snapshot test for the HomePage  component
test('HomePage  component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <HomePage  />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
