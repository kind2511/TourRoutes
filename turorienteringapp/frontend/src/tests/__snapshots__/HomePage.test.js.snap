// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage  component snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-page"
  >
    <div
      class="left"
    >
      <div
        class="sentence-section"
      >
        <p />
      </div>
    </div>
    <div
      class="right"
    >
      <a
        class="nav-logo"
        href="/"
      >
        TurRuter
      </a>
      <div
        style="display: flex; justify-content: center; margin-top: 20px;"
      >
        <button>
          Login
        </button>
        <button>
          Register
        </button>
      </div>
      <div
        class="footer-about"
      >
        <a
          href="/about"
        >
          About Us
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
